### PR TITLE
feat: view document source popup

### PR DESCRIPTION
**Pull request by kptdobe**
New sidekick feature: if you have the sidekick installed (it does not need to be opened) and you open a page with the `?view-doc-source=true` query parameter, a popup will be opened with a quick preview ("gdoc like") of the page source:

image

@rofe Before writing the tests, just want to have you overall feedback on the feature.

### CODE REVIEW

File: src/extension/background.js [hunk 1]
```diff
@@ -255,6 +255,30 @@ async function updateHelpContent() {
...
+    if (!tab.url) return;
+    try {
+      const u = new URL(tab.url);
+      const vds = u.searchParams.get('view-doc-source');
+      if (vds && vds === 'true') {
+        chrome.windows.create({
+          url: chrome.runtime.getURL(`/view-source/index.html?tabId=${id}`),
+          type: 'popup',
+          width: 740,
+          height: 1200,
```
Comment (rofe) [inline src/extension/background.js:273]: Would not specifying a `height` default to the parent window's height? Might be safer...

File: src/extension/background.js [hunk 2]
```diff
@@ -255,6 +255,30 @@ async function updateHelpContent() {
...
+/**
+ * Checks if the view source popp needs to be openeded, and opens it if necessary.
+ * @param {*} id The tab ID
+ */
+function checkViewSource(id) {
+  chrome.tabs.get(id, (tab = {}) => {
+    if (!tab.url) return;
+    try {
+      const u = new URL(tab.url);
+      const vds = u.searchParams.get('view-doc-source');
```
Comment (rofe) [inline src/extension/background.js:267]: Why is the parameter not just called `view-source`?

**Review by rofe — APPROVED** [review]

File: src/extension/background.js (hunk 2 above)
Comment (kptdobe) [reply src/extension/background.js:267]: To not conflict with the "real" view source. It is not the source of the page but the source document.

Comment (rofe) [reply src/extension/background.js:267]: Should we also rename the folder to `view-doc-source` then?

Comment (kptdobe) [reply src/extension/background.js:267]: ok, I see the confusion. I thought the folder name does not matter and we would know the difference. But I am going to rename it.

File: src/extension/background.js (hunk 1 above)
Comment (kptdobe) [reply src/extension/background.js]: Good point. I have removed the height and the browser uses the height of the current browser window.

File: src/extension/_locales/de/messages.json [hunk 3]
```diff
@@ -221,5 +221,8 @@
   },
   "help_user_status_description": {
     "message": "Markierte Themen werden nicht mehr angezeigt. Entferne das Häkchen neben einem Thema, das du noch einmal sehen möchtest."
+  },
+  "open_view_doc_source": {
+    "message": "Quelldokument der Seite anzeigen"
```
Comment (rofe) [inline src/extension/_locales/de/messages.json:226]:
```suggestion
    "message": "Quelldokument anzeigen"
```

**Review by rofe — APPROVED** [review]

**Pull request merged**